src/sentencepiece_processor.cc

```diff
@@ -600,7 +600,37 @@ util::Status SentencePieceProcessor::Decode(
   RETURN_IF_ERROR(ProcessBytePieces(byte_start, spt->pieces_size()));
 
   if (denormalizer_) {
-    *text = denormalizer_->Normalize(*text);
+    std::string normalized;
```

Comment:
This piece of code is screwing up the byte fallback pieces during decode. Needs some twisting around.